File: ruby/xortag.rb
require 'net/http'
require 'json'
require 'uri'

class XorTag
	
	def move
		register 
		
		puts "You successfully registered!"
		puts "Your player name is #{@world["name"]} and your id is #{@world["id"]}"
		
		while true do
			case rand(4)
			when 0
				move_up 
			when 1
				move_right
			when 2
				move_down
			when 3
				move_left
			end
		end
	end

	def move_up 
		uri = URI.parse(@baseUrl + "/moveup/#{@world["id"]}")
		@world = send uri
	end
	
	def move_right
		uri = URI.parse(@baseUrl + "/moveright/#{@world["id"]}")
		@world = send uri
	end
	
	def move_down
		uri = URI.parse(@baseUrl + "/movedown/#{@world["id"]}")
		@world = send uri
	end
	
	def move_left
		uri = URI.parse(@baseUrl + "/moveleft/#{@world["id"]}")
		@world = send uri
	end

	def register 
		uri = URI.parse(@baseUrl + '/register')
		@world = send uri
	end

	def send uri
		sleep(1) #Requests more frequent than once per second will fail.
		response = Net::HTTP.get(uri)
		return JSON.parse(response)
	end
	
	def initialize
		@baseUrl = 'https://xortag.azurewebsites.net'
	end

end

player = XorTag.new
player.move

File: node/xortag.js
var https = require('https');
var baseUrl = 'https://xortag.azurewebsites.net/';

var move = function(){
	switch(Math.floor((Math.random()*4))){
		case 0:
			moveUp();
		break;
		case 1:
			moveDown();
		break;
		case 2:
			moveLeft();
		break;
		case 3:
			moveRight();
		break;
	}
};

var world;

var moveUp = function(){
	request("moveup/" + world.id, move);
};

var moveDown = function(){
	request("movedown/" + world.id, move);
};

var moveLeft = function(){
	request("moveleft/" + world.id, move);
};

var moveRight = function(){
	request("moveright/" + world.id, move);
};

var look = function(){
	request("look/" + world.id, move);
};

var register = function(){
	request("register", function(){
		console.log("You successfully registered");
		console.log("Your player name is " + world.name + " and your id is " + world.id);
		move();
	});
};

var getUpdate = function(onResult) {
	return function(response){
			var str = '';
		response.on('data', function(chunk){
			str += chunk
		});
		response.on('end', function(){
			world = JSON.parse(str);
			if (onResult) onResult();
		});
	};
};

var request = function(url, onResult){
	setTimeout(function(){
		url = baseUrl + url;
		var req = https.get(url, getUpdate(onResult));
		req.end();
	}, 1000); //Requests more frequent that once per second will fail.
};

process.on('SIGINT', () => {
	process.exit();
});

register();